frontend/benefit/applicant/src/components/applications/forms/application/step1/utils/validation.ts

```diff
@@ -58,16 +61,35 @@ export const getValidationSchema = (t: TFunction): Yup.SchemaOf<Step1> =>
       )
       .max(MAX_SHORT_STRING_LENGTH, t(VALIDATION_MESSAGE_KEYS.STRING_MAX))
       .required(t(VALIDATION_MESSAGE_KEYS.REQUIRED)),
-    [APPLICATION_FIELDS_STEP1_KEYS.ORGANIZATION_TYPE]: Yup.mixed().oneOf(
-      [null, ...Object.values(ORGANIZATION_TYPES)],
-      t(VALIDATION_MESSAGE_KEYS.INVALID)
-    ),
     [APPLICATION_FIELDS_STEP1_KEYS.ASSOCIATION_HAS_BUSINESS_ACTIVITIES]:
       Yup.boolean()
         .nullable()
-        .when(APPLICATION_FIELDS_STEP1_KEYS.ORGANIZATION_TYPE, {
-          is: ORGANIZATION_TYPES.ASSOCIATION,
-          then: Yup.boolean().required(t(VALIDATION_MESSAGE_KEYS.REQUIRED)),
+        .test({
+          message: t(VALIDATION_MESSAGE_KEYS.REQUIRED),
+          test: (val) => {
+            if (
```

Comment:
kinda hard to follow the logic. Does this mean that there is for example radiobutton input for `ASSOCIATION_HAS_BUSINESS_ACTIVITIES` but only if organization type is `ASSOCIATION` ? And in this case it is required to select true or false value for the radiobutton? With other organization types there is no validation (perhaps no input either) and it always return true. 
yup can be quite complex sometimes...

Reply:
Exactly how you described it 👌